spec/PeriodicBackgroundSync-index.bs

```diff
@@ -193,3 +202,47 @@ The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn><
+    required DOMString tag;
+};
+
+[
+    Constructor(DOMString type, PeriodicSyncEventInit init),
+    Exposed=ServiceWorker
+] interface PeriodicSyncEvent : ExtendableEvent {
+    readonly attribute DOMString tag;
+  };
+</script>
+
+The {{PeriodicSyncEvent}} interface represents a [=firing=] [=periodicsync registration=].
+
+### Firing periodicSync events ### {#firing-periodicsync-events}
+Whenever the user agent changes to [=online=], it SHOULD [=fire a periodicSync event=] for each [=periodicsync registration=] whose [=registration state=] is [=pending=] and [=time to fire=] is now or in the past.
+
+### [=Fire a periodicSync event=] ### {#firing-a-periodicsync-event}
+The user agent will fire a [=periodicSync event=] for a [=periodicsync registration=] as soon as network connectivity is available, at or after the [=time to fire=] of the [=periodicsync registration=].
+If a [=periodicSync event=] fails, the user agent MAY <dfn>retry</dfn> it one or more times at a time of its choosing, based on some user agent defined heuristics.
+
+A user agent MAY impose a time limit on the lifetime extension and execution time of a {{PeriodicSyncEvent}} which is stricter than the time limit imposed for {{ExtendableEvent}}s in general. In particular, any retries of the {{PeriodicSyncEvent}} MAY have a significantly shortened time limit.
+
+To <dfn>fire a periodicSync event</dfn> for a [=periodicsync registration=] <var>registration</var>, the user agent MUST run the following steps:
+1. [=Assert=]: |registration|'s [=registration state=] is [=pending=].
+1. Let <var>serviceWorkerRegistration</var> be the [=service worker registration=] associated with |registration|.
+1. [=Assert=]: |registration|'s [=time to fire=] is equal to the current time or in the past.
+1. Set |registration|'s [=registration state=] to [=firing=].
+1. [=Fire functional event=] "<code>periodicSync</code>" using {{PeriodicSyncEvent}} on |serviceWorkerRegistration| with {{PeriodicSyncEvent/tag}} set to the [=tag=] associated with |registration|.
+1. Then Run the following steps with <var>dispatchedEvent</var>:
+    1. Let <var>waitUntilPromise</var> be the result of [=waiting for all=] of |dispatchedEvent|'s [=extend lifetime promises=].
+    1. [=Upon fulfillment=] of |waitUntilPromise|, perform the following steps atomically:
+        1. Set |registration|'s state to [=pending=].
+        1. [=Calculate a time to fire=] for |registration|.
+        1. [=Schedule processing=] for |registration|.
+        1. Abort the rest of these steps.
+    1. [=Upon rejection=] of |waitUntilPromise|, or if the script has been aborted by the [=Terminate Service Worker|termination=] of the [=service worker=], perform the following steps atomically:
+        1. Set |registration|'s state to [=pending=].
+        1. [=Calculate a time to fire=] for |registration|.
+        1. [=Schedule processing=] for |registration|.
+        1. Abort the rest of these steps.
```

Comment:
There are no further steps? Dito on :243 above, if :244 were phrased as an else

Reply:
Remove this.
There's no need for Else IMO because there's no If, and only one of the two will happen -- the promise will either complete or be rejected. Let me know if you disagree.